### PR TITLE
Add metacardLocation

### PR DESCRIPTION
**Pull request by emmberk**
This PR is ready for review. This PR splits out a commit from https://github.com/connexta/ion-query-api/pull/36.
```java
/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech) (4.2.2).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
package com.connexta.search.rest.spring;

import com.connexta.search.rest.models.ErrorMessage;
import com.connexta.search.rest.models.Result;
import io.swagger.annotations.*;
import org.springframework.http.HttpStatus;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestHeader;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RequestPart;
import org.springframework.web.context.request.NativeWebRequest;
import org.springframework.web.multipart.MultipartFile;

import javax.validation.Valid;
import javax.validation.constraints.*;
import java.util.List;
import java.util.Map;
import java.util.Optional;
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2019-12-02T11:46:08.014561-07:00[America/Phoenix]")

@Validated
@Api(value = "Query", description = "the Query API")
public interface QueryApi {

    default Optional<NativeWebRequest> getRequest() {
        return Optional.empty();
    }

    @ApiOperation(value = "Query endpoint for datasets.", nickname = "query", notes = "A system can use the Query endpoint to perform a query on datasets. Results are returned in order from most matched to least matched. ", response = Result.class, responseContainer = "List", tags={ "query", })
    @ApiResponses(value = { 
        @ApiResponse(code = 200, message = "The query request was successful. ", response = Result.class, responseContainer = "List"),
        @ApiResponse(code = 400, message = "The client message could not understood by the server due to invalid format or syntax. ", response = ErrorMessage.class),
        @ApiResponse(code = 401, message = "The client could not be authenticated. ", response = ErrorMessage.class),
        @ApiResponse(code = 403, message = "The client does not have permission. ", response = ErrorMessage.class),
        @ApiResponse(code = 501, message = "The requested API version is not supported and therefore not implemented. ", response = ErrorMessage.class),
        @ApiResponse(code = 200, message = "Any other possible errors not currently known. ", response = ErrorMessage.class) })
    @RequestMapping(value = "/search",
        produces = { "application/json" }, 
        method = RequestMethod.GET)
    default ResponseEntity<List<Result>> query(@NotNull @Pattern(regexp=".*\\S.*") @Size(min=1,max=5000) @ApiParam(value = "The OGC CommonQL string that will be used in the query (see the \"Annex B - BNF Definition of OGC CommonQL (Normative)\" section at http://docs.opengeospatial.org/is/12-168r6/12-168r6.html). Supported attribute names and boolean terms should be defined by the implementing server. ", required = true) @Valid @RequestParam(value = "q", required = true) String q) {
        getRequest().ifPresent(request -> {
            for (MediaType mediaType: MediaType.parseMediaTypes(request.getHeader("Accept"))) {
                if (mediaType.isCompatibleWith(MediaType.valueOf("application/json"))) {
                    String exampleString = "{ \"irmLocation\" : \"https://openapi-generator.tech\", \"metacardLocation\" : \"https://openapi-generator.tech\" }";
                    ApiUtil.setExampleResponse(request, "application/json", exampleString);
                    break;
                }
            }
        });
        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);

    }

}
```
```java
package com.connexta.search.rest.models;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.net.URI;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Result
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2019-12-02T11:45:55.539568-07:00[America/Phoenix]")

public class Result   {
  @JsonProperty("metacardLocation")
  private URI metacardLocation;

  @JsonProperty("irmLocation")
  private URI irmLocation;

  public Result metacardLocation(URI metacardLocation) {
    this.metacardLocation = metacardLocation;
    return this;
  }

  /**
   * Get metacardLocation
   * @return metacardLocation
  */
  @ApiModelProperty(required = true, value = "")
  @NotNull

  @Valid

  public URI getMetacardLocation() {
    return metacardLocation;
  }

  public void setMetacardLocation(URI metacardLocation) {
    this.metacardLocation = metacardLocation;
  }

  public Result irmLocation(URI irmLocation) {
    this.irmLocation = irmLocation;
    return this;
  }

  /**
   * Get irmLocation
   * @return irmLocation
  */
  @ApiModelProperty(required = true, value = "")
  @NotNull

  @Valid

  public URI getIrmLocation() {
    return irmLocation;
  }

  public void setIrmLocation(URI irmLocation) {
    this.irmLocation = irmLocation;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    Result result = (Result) o;
    return Objects.equals(this.metacardLocation, result.metacardLocation) &&
        Objects.equals(this.irmLocation, result.irmLocation);
  }

  @Override
  public int hashCode() {
    return Objects.hash(metacardLocation, irmLocation);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class Result {\n");
    
    sb.append("    metacardLocation: ").append(toIndentedString(metacardLocation)).append("\n");
    sb.append("    irmLocation: ").append(toIndentedString(irmLocation)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
```
```java
package com.connexta.search.rest.models;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Error message information.
 */
@ApiModel(description = "Error message information.")
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2019-12-02T11:45:55.539568-07:00[America/Phoenix]")

public class ErrorMessage   {
  @JsonProperty("timestamp")
  private OffsetDateTime timestamp;

  @JsonProperty("path")
  private String path;

  @JsonProperty("status")
  private Integer status;

  @JsonProperty("code")
  private Integer code;

  @JsonProperty("error")
  private String error;

  @JsonProperty("message")
  private String message;

  @JsonProperty("details")
  @Valid
  private List<String> details = null;

  public ErrorMessage timestamp(OffsetDateTime timestamp) {
    this.timestamp = timestamp;
    return this;
  }

  /**
   * The server time when the error was generated.
   * @return timestamp
  */
  @ApiModelProperty(required = true, value = "The server time when the error was generated.")
  @NotNull

  @Valid

  public OffsetDateTime getTimestamp() {
    return timestamp;
  }

  public void setTimestamp(OffsetDateTime timestamp) {
    this.timestamp = timestamp;
  }

  public ErrorMessage path(String path) {
    this.path = path;
    return this;
  }

  /**
   * The path for the REST request that generated this error.
   * @return path
  */
  @ApiModelProperty(example = "/search", required = true, value = "The path for the REST request that generated this error.")
  @NotNull


  public String getPath() {
    return path;
  }

  public void setPath(String path) {
    this.path = path;
  }

  public ErrorMessage status(Integer status) {
    this.status = status;
    return this;
  }

  /**
   * The corresponding HTTP status code.
   * @return status
  */
  @ApiModelProperty(example = "200", required = true, value = "The corresponding HTTP status code.")
  @NotNull


  public Integer getStatus() {
    return status;
  }

  public void setStatus(Integer status) {
    this.status = status;
  }

  public ErrorMessage code(Integer code) {
    this.code = code;
    return this;
  }

  /**
   * An optional more specific code corresponding to the error (specific codes need to be documented). 
   * @return code
  */
  @ApiModelProperty(value = "An optional more specific code corresponding to the error (specific codes need to be documented). ")


  public Integer getCode() {
    return code;
  }

  public void setCode(Integer code) {
    this.code = code;
  }

  public ErrorMessage error(String error) {
    this.error = error;
    return this;
  }

  /**
   * The corresponding HTTP reason phrase.
   * @return error
  */
  @ApiModelProperty(example = "Bad Request", required = true, value = "The corresponding HTTP reason phrase.")
  @NotNull

@Size(min=1) 
  public String getError() {
    return error;
  }

  public void setError(String error) {
    this.error = error;
  }

  public ErrorMessage message(String message) {
    this.message = message;
    return this;
  }

  /**
   * Specific information about the error.
   * @return message
  */
  @ApiModelProperty(example = "Validation failed for object='systemInfo'.", required = true, value = "Specific information about the error.")
  @NotNull

@Size(min=1) 
  public String getMessage() {
    return message;
  }

  public void setMessage(String message) {
    this.message = message;
  }

  public ErrorMessage details(List<String> details) {
    this.details = details;
    return this;
  }

  public ErrorMessage addDetailsItem(String detailsItem) {
    if (this.details == null) {
      this.details = new ArrayList<>();
    }
    this.details.add(detailsItem);
    return this;
  }

  /**
   * Optional detailed information about the error.
   * @return details
  */
  @ApiModelProperty(value = "Optional detailed information about the error.")


  public List<String> getDetails() {
    return details;
  }

  public void setDetails(List<String> details) {
    this.details = details;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ErrorMessage errorMessage = (ErrorMessage) o;
    return Objects.equals(this.timestamp, errorMessage.timestamp) &&
        Objects.equals(this.path, errorMessage.path) &&
        Objects.equals(this.status, errorMessage.status) &&
        Objects.equals(this.code, errorMessage.code) &&
        Objects.equals(this.error, errorMessage.error) &&
        Objects.equals(this.message, errorMessage.message) &&
        Objects.equals(this.details, errorMessage.details);
  }

  @Override
  public int hashCode() {
    return Objects.hash(timestamp, path, status, code, error, message, details);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ErrorMessage {\n");
    
    sb.append("    timestamp: ").append(toIndentedString(timestamp)).append("\n");
    sb.append("    path: ").append(toIndentedString(path)).append("\n");
    sb.append("    status: ").append(toIndentedString(status)).append("\n");
    sb.append("    code: ").append(toIndentedString(code)).append("\n");
    sb.append("    error: ").append(toIndentedString(error)).append("\n");
    sb.append("    message: ").append(toIndentedString(message)).append("\n");
    sb.append("    details: ").append(toIndentedString(details)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
```

### CODE REVIEW

File: rest/specs/src/main/filtered-resources/openapi/openapi.yaml [hunk 1]
```diff
@@ -11,8 +11,7 @@ info:
   version: ${project.version}
   title: Query API
   description: >
-    The Query API is defined to provide clients with the ability to
-    query for stored datasets.
+    The Query API is defined to provide clients with the ability to query on datasets.
```
Comment (figliold) [inline rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: ❓ Is this true?  This makes it sound like we return a list of datasets, but we really return a list of Results (metacardLocation and IrmLocation).

Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: Is says query on datasets. It does not say that it returns datasets.

Comment (figliold) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: Maybe it's just me, but when I query on "something", I would expect to get a list of that "something" back. 🤷‍♂

Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: Do you have a suggestion for a better description?

Comment (figliold) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: I'm just saying that if we are going to use datasets in our system, we should return datasets not parts of datasets.  It looks like a bad design to me, but maybe we are just stuck with it.

Comment (emmberk) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: Do you agree that updating the API to return datasets is outside the scope of this PR? If yes, do you think this description needs to be updated in this PR?

Comment (figliold) [reply rest/specs/src/main/filtered-resources/openapi/openapi.yaml]: I agree that updating the API to return datasets is outside the scope of this PR.  I guess we can leave the description as-is.

**Pull request merged**